### PR TITLE
[k8s-extension] Release v0.7.0

### PR DESCRIPTION
**Pull request by jonathan-innis**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
k8s-extension

File: src/k8s-extension/azext_k8s_extension/commands.py [hunk 1]
```diff
@@ -20,6 +20,6 @@ def load_command_table(self, _):
                             is_preview=True) \
             as g:
         g.custom_command('create', 'create_k8s_extension')
-        g.custom_command('delete', 'delete_k8s_extension', confirmation=True)
+        g.custom_command('delete', 'delete_k8s_extension')
```
Comment (zhoxing-ms) [inline src/k8s-extension/azext_k8s_extension/commands.py:23]: May I ask why do you delete the confirmation?

File: src/k8s-extension/azext_k8s_extension/partner_extensions/DefaultExtension.py [hunk 2]
```diff
@@ -56,5 +58,11 @@ def Update(self, extension, auto_upgrade_minor_version, release_train, version):
...
-    def Delete(self, client, resource_group_name, cluster_name, name, cluster_type):
-        pass
+    def Delete(self, cmd, client, resource_group_name, cluster_name, name, cluster_type, yes):
+        user_confirmation_factory(cmd, yes)
+
+
+def user_confirmation_factory(cmd, yes, message="Are you sure you want to perform this operation?"):
+    if cmd.cli_ctx.config.getboolean('core', 'disable_confirm_prompt', fallback=False):
+        return
+    user_confirmation(message, yes=yes)
```
Comment (zhoxing-ms) [inline src/k8s-extension/azext_k8s_extension/partner_extensions/DefaultExtension.py:68]: Why do we need a config to turn off confirmation?

Comment (jonathan-innis) [reply src/k8s-extension/azext_k8s_extension/partner_extensions/DefaultExtension.py:68]: This is an azure core config concept. If that someone has done `az config set core.disable_confirm_prompt=true`, we want to honor this as well for the user_confirmations that we create

File: src/k8s-extension/azext_k8s_extension/commands.py (hunk 1 above)
Comment (jonathan-innis) [reply src/k8s-extension/azext_k8s_extension/commands.py:23]: We want to allow custom logic for our partners to have confirmations or not and to print different kinds of messages based on the extension_type. We have moved the confirmation logic into the `user_confirmation_factory`

Comment (zhoxing-ms) [reply src/k8s-extension/azext_k8s_extension/commands.py:23]: Got it, thanks~

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**